### PR TITLE
feat: extractLsBytes and related proofs for simp_mem [1/3?]

### PR DESCRIPTION
**Pull request by bollu**
This PR peels the bitvector library features from
https://github.com/leanprover/LNSym/pull/90/,
with theorems for `extractLsBytes`.
This enables us to rewrite proofs about reading and writing bytes in memory with a higher level API.

### License:

By submitting this pull request, I confirm that my contribution is
made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: Arm/BitVec.lean [hunk 1]
```diff
@@ -968,13 +1009,63 @@ def extractLsBytes (val : BitVec w) (base : Nat) (n : Nat) : BitVec (n * 8) :=
...
+      congr 1
+      omega
+    · simp only [h, decide_False, Bool.false_and, decide_True, Bool.true_and, Bool.false_eq]
+      apply BitVec.getLsb_ge
+      omega
+  · simp only [hn, ↓reduceIte, getLsb_zero, Bool.and_eq_false_imp, decide_eq_true_eq]
+    omega
+
+/-- Extracting out bytes from the zero bitvector is equal to the zero bitvector. -/
+@[simp]
```
Comment (shigoel) [inline Arm/BitVec.lean:1046]: `simp` tag?

File: Arm/BitVec.lean [hunk 2]
```diff
@@ -938,13 +938,38 @@ theorem sub_le_sub_iff_right (a b c : BitVec w₁) (hac : c ≤ a)
 
 /-! ### Least Significant Byte -/
 
-/-- Definition to extract the `n`th least significant *Byte* from a bitvector. -/
+/--
+Definition to extract the `n`th least significant *Byte* from a bitvector.
+TODO: this should be named `getLsByte`.
```
[the remark this answers is not in the document]
Comment (bollu) [reply Arm/BitVec.lean:943]: @shigoel The definition already exists, so I'd recommend renaming it in a different PR, if we want to.

[the remark this answers is not in the document]
Comment (shigoel) [reply Arm/BitVec.lean:943]: Sounds good to me! `extractLsBytes` is painful for me to type. :(

**Comment by bollu** [comment]
@shigoel All comments addressed.

**Review by shigoel — APPROVED** [review]
Thanks!

**Pull request merged**